docs/install.sh

```diff
@@ -26,44 +25,67 @@ fi
 
 echo "INFO: verifying cpu architecture compatibility"
 
-ARCH=$(uname -m)
-ARCH=$(echo "${ARCH,,}") # convert to lower case
+ARCH=$(uname -m | tr '[:upper:]' '[:lower:]')
 
 if [ "$ARCH" = x86_64 ]; then
     ARCH="amd64"
 fi
+if [ "$ARCH" = aarch64 ]; then
+    ARCH="arm64"
+fi
 
-if [ "$ARCH" != "amd64" ]; then
+if [[ "$ARCH" != "amd64" && "$ARCH" != "arm64" ]]; then
     echo "Error: ${ARCH} is not a supported architecture at this time."
     exit 1
 fi
 
-# ------------------------------------------------------------------------------
-# Determine Latest Release
-# ------------------------------------------------------------------------------
+TEMPDIR=$(mktemp -d)
 
-LATEST_RELEASE=$(curl -s https://api.github.com/repos/${BASE}/releases/latest | perl -ne 'print $1 if m{"name": "(v[0-9]+\.[0-9]+\.[0-9]+.*)"}')
+function download_with_curl()
+{
+    echo "INFO: downloading ktf via curl"
 
-if [ "$LATEST_RELEASE" = "" ]; then
-    echo "Error: could not find latest release for ${BASE}!${LATEST_RELEASE}"
-    exit 1
-fi
+    # ------------------------------------------------------------------------------
+    # Determine Latest Release
+    # ------------------------------------------------------------------------------
 
-echo "INFO: the latest release of ${BASE} was determined to be ${LATEST_RELEASE}"
+    LATEST_RELEASE=$(curl -s https://api.github.com/repos/${BASE}/releases/latest | perl -ne 'print $1 if m{"name": "(v[0-9]+\.[0-9]+\.[0-9]+.*)"}')
```

Comment:
there's already a pattern for tool download in the Makefile:


https://github.com/Kong/kubernetes-ingress-controller/blob/4187223cfffce38fb0a6691a565534a0176f8711/Makefile#L34

Reply:
It is indeed. But the purpose of this PR was not to change this mechanism but rather introduce a new way of downloading via `gh`: which download release artifacts, not builds from source via `go install ...`.

We might think about this as a fallback but then what would be the point of this script if it was an equivalent of calling `go install github.com/kong/kubernetes-testing-framework/cmd/ktf@latest` 🤔